src/Common/src/Interop/Unix/libssl/ApplicationProtocolContext.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net
+{
+    internal enum ApplicationProtocolNegotiationStatus
+    {
+        None = 0,
+        Success,
+        SelectedClientOnly
+    }
+
+    internal enum ApplicationProtocolNegotiationExt
+    {
+        None = 0,
+        NPN,
+        ALPN
+    }
+
+    internal class ApplicationProtocolContext
+    {
+        public ApplicationProtocolNegotiationStatus NegotiationStatus = ApplicationProtocolNegotiationStatus.None;
+        public ApplicationProtocolNegotiationExt NegotiationExtension = ApplicationProtocolNegotiationExt.None;
+
+        public string GetProtocolId()
+        {
+            return null;
```

Comment:
Ideally we'd implement it at the same time.  If that's not possible, then yes, opening an issue to track the Unix implementation, and including a TODO comment here citing that issue number.  Then there's a question of what the temporary behavior here should be.  I've not yet seen how this is consumed, but often the right behavior is "throw new PlatformNotSupportedException();"... but it depends on the usage.